src/main/java/core/basesyntax/service/impl/MapTransactionService.java

```
@@ -0,0 +1,18 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.TransactionType;
+import core.basesyntax.service.TransactionHandler;
+import core.basesyntax.service.TransactionService;
+import java.util.Map;
+
+public class MapTransactionService implements TransactionService {
+    private final Map<TransactionType, TransactionHandler> handlers;
+
+    public MapTransactionService(Map<TransactionType, TransactionHandler> handlers) {
+        this.handlers = Map.copyOf(handlers);
+    }
+
+    public TransactionHandler selectHandler(TransactionType transactionType) {
+        return handlers.get(transactionType);
```

`selectHandler` returns `handlers.get(transactionType)` and may return null when there is no handler for the requested transaction type. This violates the requirement: "Validate operation codes are known; if not, throw informative RuntimeException (e.g., incorrect strategy)." Instead of returning null, validate `transactionType` (null-check) and throw a RuntimeException with a message such as: "No handler registered for transaction type: " + transactionType + ". Check strategy map configuration." This will make misconfiguration actionable rather than causing NPEs later.